Description: Use lsof to count open fds before setting ulimit.
 One of the Debian alpha buildds and several of the Ubuntu buildds hit the fd
 ulimit that the testsuite is run under.  Presumably there are extra fds open
 in the buildd process for some reason.  So log what fds are open (for
 debugging purposes) and set the limit by adding to this number.
Author: Olly Betts <olly@survex.com>
Index: xapian-core/tests/runtest.in
===================================================================
--- xapian-core/tests/runtest.in	(revision 13449)
+++ xapian-core/tests/runtest.in	(working copy)
@@ -35,7 +35,13 @@
 # If ulimit supports it, limit the number of open file descriptors so we catch
 # file descriptor leaks sooner.  Not all platforms support changing this limit
 # but Linux does at least.
-ulimit -n 64 2>/dev/null || true
+echo
+echo "lsof reports open fds:"
+lsof -d0-1023 -a -p$$
+echo
+pids_already_used=`lsof -d0-32767 -a -p$$ 2>/dev/null|wc -l`
+# One for the header line, one for the fd used for the `...`.
+ulimit -n `expr $pids_already_used - 2 + 64` 2>/dev/null || true
 
 case $1 in
   *test|*test@EXEEXT@) ;;
